src/utils.jl

```diff
@@ -13,6 +13,71 @@ See also: [`imfilter`](@ref).
 """
 centered(A::AbstractArray) = OffsetArray(A, map(n->-((n+1)>>1), size(A)))
 
+"""
+    kernfft = freqkernel([T::Type], kern, sz=size(kern); rfft=false)
+
+Return a frequency-space representation of `kern`.
+This embeds `kern` in an array of size `sz`, in a manner that implicitly imposes periodic boundary
+conditions, and then returns the fourier transform. This is sometimes called the optical transfer
+function, and known in some frameworks as `psf2otf`. If `rfft` is `true`, the fft for real-valued
+arrays (`rfft`) is returned instead and the first dimension size will be approximately half of `sz[1]`.
+
+`kern` should be zero-centered, i.e., `kern[0, 0]` should reference the center of your kernel.
+See [`centered`](@ref). Optionally specify the numeric type `T` (which must be one of the
+types supported by FFTW, either `Float32` or `Float64`).
+
+The inverse of `freqkernel` is [`spacekernel`](@ref).
+"""
+function freqkernel(::Type{T}, kern::AbstractArray, sz::Dims=size(kern); rfft=false) where T<:Union{Float32,Float64}
```

Comment:
Since `T` is float type, will it be reasonable to use `rfft` by default? Or is there a need to use `fft` rather than `rfft` here?

Reply:
Great question. While it's not done here, one could implement `freqkernel` for complex-valued kernels. If we make `rfft` the default it will have to be a different API, but with `fft` the default then the generalization is basically trivial.